src/basic/mkdir.h

```diff
@@ -25,6 +25,7 @@
 
 typedef enum MkdirFlags {
         MKDIR_FOLLOW_SYMLINK = 1 << 0,
+        MKDIR_WARN_MODE      = 1 << 1,
```

Comment:
this does more than "warn", right? it also refuses, hence maybe call thus MKDIR_REFUSE_AND_WARN_MODE or so?

Reply:
Or MKDIR_PEDANTIC? MKDIR_FASCIST? MKDIR_THROW_TANTRUM? ;-)

Reply:
also, it checks more than just the mode. hence maybe MKDIR_COMPLAIN_STRICT_PERMS?

Reply:
or maybe split this up: MKDIR_WARN as one flag, and MKDIR_STRICT_PERMISSIONS as another one? And then doe the usual `log_full(flags & MKDIR_WARN ? LOG_ERR : LOG_DEBUG, …` dance? I figure that would be the most systematic approach?

Reply:
No, this flag only influences logging. (At least that was the intent. But looking at the code, it seems correct.)

I'll rework the patch to use `log_full` though.